client/components/filters/advanced/search-filter.js

```diff
@@ -41,39 +42,65 @@ class SearchFilter extends Component {
 		onFilterChange( filter.key, 'value', idList );
 	}
 
+	getLegend( filter, config ) {
+		const { selected } = this.state;
+		const rule = find( config.rules, { value: filter.rule } ) || {};
+		const filterStr = selected.map( item => item.label ).join( ', ' );
+
+		return interpolateComponents( {
+			mixedString: config.labels.title,
+			components: {
+				filter: <span>{ filterStr }</span>,
+				rule: <span>{ rule.label }</span>,
+			},
+		} );
+	}
+
 	render() {
-		const { config, filter, instanceId, onFilterChange } = this.props;
+		const { config, filter, onFilterChange, isEnglish } = this.props;
 		const { selected } = this.state;
 		const { key, rule } = filter;
 		const { input, labels, rules } = config;
-		return (
-			<Fragment>
-				<div
-					id={ `${ key }-${ instanceId }` }
-					className="woocommerce-filters-advanced__fieldset-legend"
-				>
-					{ labels.title }
-				</div>
-				{ rule && (
+		const children = interpolateComponents( {
+			mixedString: labels.title,
+			components: {
+				rule: (
 					<SelectControl
-						className="woocommerce-filters-advanced__list-specifier"
+						className="woocommerce-filters-advanced__rule"
 						options={ rules }
 						value={ rule }
 						onChange={ partial( onFilterChange, key, 'rule' ) }
 						aria-label={ labels.rule }
 					/>
-				) }
-				<div className="woocommerce-filters-advanced__list-selector">
+				),
+				filter: (
 					<Search
+						className="woocommerce-filters-advanced__input"
 						onChange={ this.onSearchChange }
 						type={ input.type }
 						placeholder={ labels.placeholder }
 						selected={ selected }
-						ariaLabelledby={ `${ key }-${ instanceId }` }
+						aria-label={ labels.filter }
```

Comment:
Search doesn't support aria-label at the moment, but that can also be fixed in another PR

Reply:
And now it does #484, so you can ignore this comment 😁